daemons/compute/client-go/main.go

```diff
@@ -99,7 +102,18 @@ func main() {
 				break
 			}
 
+			// Cancel the data movement after some iterations
+			if *cancelExpiryTime >= 0 && count >= *cancelExpiryTime {
+				log.Printf("Canceling request: %v", uid)
+				cancelResponse, err := cancelRequest(ctx, c, *workflow, *namespace, uid)
+				if err != nil {
+					log.Fatalf("error initiating data movement cancel request: %v", err)
+				}
+				log.Printf("Data movement request cancel initiated: %v %v", uid, cancelResponse.String())
+			}
+
 			time.Sleep(time.Second)
```

Comment:
Instead of second counting, why not just sleep the desired time? You could do that before the getStatus() call

Reply:
In the normal case (without cancel) there is no desired time. The loop just prints out the status of the DM every second until it's completed.

Reply:
if cancelExpiryTime is non-negative, you would wait desired time and then issue the cancel
then, in all cases, you'd poll for completion

Reply:
Got it. I like it. Updated.